spec/lib/tasks/db_data_migrate_rake_spec.rb

```diff
@@ -141,5 +141,66 @@
         expect(openid_authentication_service).to be_a OpenidAuthenticationService
       }
     end
+
+    shared_examples 'a consistency migration' do |prep_method_sym|
+      let(:record_class) { record.class }
+      before do
+        expect(record).to be_persisted
+      end
+      context 'record is consistent' do
+        let(:prep_method) { send(prep_method_sym) }
+        before do
+          prep_method.call
+        end
+        it 'should update is_consistent to true' do
+          expect(record_class.where(is_consistent: nil)).to exist
+          invoke_task
+          expect(record_class.where(is_consistent: nil)).not_to exist
+          record.reload
+          expect(record.is_consistent).to eq true
+        end
+      end
+
+      context 'record is not consistent' do
+        it 'should update is_consistent to false' do
+          invoke_task
+          record.reload
+          expect(record.is_consistent).to eq false
+        end
+      end
+    end
+
+    describe 'consistency migration', :vcr do
+      let(:storage_provider) { FactoryGirl.create(:storage_provider, :swift) }
+
+      before do
+        expect(storage_provider).to be_persisted
+      end
+
+      context 'for project' do
+        let(:record) { FactoryGirl.create(:project, is_consistent: nil) }
+        let(:init_project_storage) {
+          Proc.new do
+            storage_provider.put_container(record.id)
+            expect(storage_provider.get_container_meta(record.id)).to be
+          end
+        }
+        it_behaves_like 'a consistency migration', :init_project_storage
+      end
+
+      context 'for upload' do
+        let(:record) { FactoryGirl.create(:upload, is_consistent: nil, storage_provider: storage_provider) }
+        let(:init_upload) {
+          Proc.new do
+            storage_provider.put_container(record.project.id)
+            expect(storage_provider.get_container_meta(record.project.id)).to be
+            record.create_and_validate_storage_manifest
+            record.update(is_consistent: nil)
```

Comment:
Be aware that using `update` will run validations. If you want to skip validations, use `update_attribute`. If you also want to skip callbacks, use `update_column`. If you don't care about either, use `update!` so that an exception is thrown if something doesn't work as expected.